source/common/http/conn_manager_utility.cc

```diff
@@ -19,13 +19,13 @@ namespace Envoy {
 namespace Http {
 
 Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequestHeaders(
-    Http::HeaderMap& request_headers, Protocol protocol, Network::Connection& connection,
+    Http::HeaderMap& request_headers, Network::Connection& connection,
     ConnectionManagerConfig& config, const Router::Config& route_config,
     Runtime::RandomGenerator& random, Runtime::Loader& runtime,
     const LocalInfo::LocalInfo& local_info) {
   // If this is a Upgrade request, do not remove the Connection and Upgrade headers,
   // as we forward them verbatim to the upstream hosts.
-  if (protocol == Protocol::Http11 && Utility::isUpgrade(request_headers)) {
+  if (Utility::isUpgrade(request_headers)) {
```

Comment:
why remove the check? the extended connect protocol does not use any upgrade headers from what I see.

Reply:
Because as docced up, for filter chain consistency, we consistently transform upgrades to H1 style headers at the codec layer.  If there's 
client - H1 upgrade - Envoy - H2 hop - another Envoy - H2 hop - something else
the Envoy sending and receiving the upgrade in H2 form will still see HTTP/1.1 style upgrade headers at the http connection manager and in the HTTP filters.

Reply:
We definitely need to include HTTP/2 here given the way we're doing upgrades in Envoy.  Upgrade is an HTTP/1.1 header so arguably we could protocol checks back in and explicitly disallow something we don't expect folks to do.  Given that 1.0 support is off by default and I don't think anyone's trying to do 1.0 upgrades I lean towards not worrying about it but I'm happy to add it back if you'd prefer!

Reply:
Ok, I'm fine with it as-is.